docs/src/pages/components/data-grid/export/export.md

```diff
@@ -57,6 +57,19 @@ There are a few ways to include or hide other columns:
 <DataGrid columns={[{ field: 'id', disableExport: true }, { field: 'brand' }]} />
 ```
 
+#### Exported rows
+
+By default, if some rows are selected, the CSV export will only contain those rows.
+If not, the CSV export will contain all the rows passing the filtering process and not collapsed.
+
+Set the `getRowsToExport` to manually pick the rows to include in the CSV export.
+
+{{"demo": "pages/components/data-grid/export/CsvGetRowsToExport.js", "bg": "inline", "defaultCodeOpen": false}}
+
+When using [Row grouping](/components/data-grid/group-pivot/#row-grouping), it can be useful to remove the groups from the CSV export
```

Comment:
The behavior of this example is not exactly "get the raw data", or the equivalent of a `disableRowGrouping` prop.
The rows are still grouped, we don't export the groups themselves, but the order of the rows respect the grouping.
With that being said, I don't know how much valuable it is.

I agree that we always have a tension between *easy to use* and *powerful*.
With the use of built-in selectors, I feel like this method is quite easy to use while being as powerful as possible.
And we could make the "remove the group headers" example even easier with a built-in `gridVisibleSortedLeafRowIdsSelector` for instance.
But if we find a good behavior with a boolean flag, it can be easier. We just have to make sure it stays consistent with the selection for instance (don't limit the export to the selected rows if we only selected groups for instance).

Reply:
Got it. `hideRowGroupHeaders` would make more sense as a suggested prop for this use case then.
But I agree that the demand for those specific props are not clear. 

In that case, what do you think about making a section on docs to talk about extending the `getRowsToExport` with the built in selectors? And briefly describe each selector instead of focusing on this RowGrouping use case? 
I mean, we can still use the RowGrouping example, but the primary focus of the section would be to explicitly show  the selectors.

Reply:
For me it was the goal of this demo: https://github.com/mui-org/material-ui-x/pull/3687/files#diff-2fe596f167a34b21fe133bf8e54b54e1a602a3a36cf40b1a13fa7fb9486a0615R67
[Doc preview](https://deploy-preview-3687--material-ui-x.netlify.app/components/data-grid/export/#custom-exported-content).

But we could add a description for each selector before, and if someday we have a better doc around these selectors (see next paragraph of this comment), we could just link to it.

This relates with #3717, we have several row selector that I unified for v5 (each time with a `gridXXXIdsSelectors` which returns the ids and `gridXXXEntriesSelectors` which returns the ids and the models).
And I feel like, it is one of the main topic to document as well as possible around selectors.

Reply:
Sounds good. Let's do it on a specific section here, and push further the docs on the selectors with [#3717](https://github.com/mui-org/material-ui-x/issues/3717)